src/detect-parse.c

```diff
@@ -2373,38 +2394,59 @@ void DetectParseFreeRegexes(void)
 
 /** \brief add regex and/or study to at exit free list
  */
-void DetectParseRegexAddToFreeList(pcre *regex, pcre_extra *study)
+void DetectParseRegexAddToFreeList(DetectParseRegex *detect_parse)
 {
     DetectParseRegex *r = SCCalloc(1, sizeof(*r));
     if (r == NULL) {
         FatalError(SC_ERR_MEM_ALLOC, "failed to alloc memory for pcre free list");
     }
-    r->regex = regex;
-    r->study = study;
+    r->regex = detect_parse->regex;
+    r->study = detect_parse->study;
     r->next = g_detect_parse_regex_list;
+    r->jit_stack = detect_parse->jit_stack;
     g_detect_parse_regex_list = r;
 }
 
-void DetectSetupParseRegexes(const char *parse_str,
-                             pcre **parse_regex,
-                             pcre_extra **parse_regex_study)
+void DetectSetupParseRegexesOptsWithJIT(const char *parse_str, DetectParseRegex *detect_parse, int opts, bool jit)
```

Comment:
I don't see how this would be used with JIT. AFAIK the JIT support does not support substring capture, which is really the only thing we use pcre for during keyword parsing.

It seems that the JIT condition is never true.